### ISSUE TITLE
When I use your code,dim(ad.count) equal to Null.Why is that?

### ISSUE THREAD

**Issue by zhukun3** [EvaYiwenWang/PLSDAbatch_workflow#7]
After install,I test your code `data('AD_data') 
ad.count <- AD_data$FullData$X.count
dim(ad.count)`,R returned me Null.By the way,one third of my peanut RNA-seq data samples have batch effect, am I suitable for this R package?
![image](https://github.com/user-attachments/assets/b1aa2c22-b922-4b85-9a1c-accefd9f15f8)

**Comment by EvaYiwenWang** [comment]
Sorry about that, I changed my data structure. Please try this:
data('AD_data') 
ad.count <- assays(AD_data$FullData)$Count
dim(ad.count)
For more information, please check here: https://www.bioconductor.org/packages/release/bioc/vignettes/PLSDAbatch/inst/doc/brief_vignette.html.

For your RNA-seq data, could you show me the design of batch and treatment by `table(batch,  treatment)`? I need to check if the design is balanced. In addition, for RNA-seq data, you don't need CLR transformation, you can use other normalisation specific for RNA-seq, then use my PLSDA-batch. 

Best, 
Eva

**Comment by zhukun3** [comment]
Thank you,Eva,for my RNA-seq data,the yellow part is the RNA-seq data measured by one enzyme, and the green part is the RNA-seq data measured by another enzyme. I want to eliminate the batch effect brought by these two enzymes. Can PLSDA-batch be used for this experimental design?
![image](https://github.com/user-attachments/assets/726b517e-84ad-427c-81bb-c3d4a9c98f8d)

**Comment by zhukun3** [comment]
> Thank you,Eva,for my RNA-seq data,the yellow part is the RNA-seq data measured by one enzyme, and the green part is the RNA-seq data measured by another enzyme. I want to eliminate the batch effect brought by these two enzymes. Can PLSDA-batch be used for this experimental design? ![image](https://private-user-images.githubusercontent.com/83392042/387473873-726b517e-84ad-427c-81bb-c3d4a9c98f8d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE5ODc4NDQsIm5iZiI6MTczMTk4NzU0NCwicGF0aCI6Ii84MzM5MjA0Mi8zODc0NzM4NzMtNzI2YjUxN2UtODRhZC00MjdjLTgxYmItYzNkNGE5Yzk4ZjhkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTE5VDAzMzkwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU4M2E3ZGRmYmM5NDhkYWZmNjc5Y2JkODljN2EyNWVkMzE5YmYxOGRhNjhhZjQ0NTE2YmUwNjJlYzBlN2VlNTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.wQm3M09bnGDxrK0YnCkZOo3v-nI0dqd1vA4A8IRks3U)

To summarize, 39 samples used one enzyme, 24 samples used another enzyme, whether the design was balanced?

**Comment by zhukun3** [comment]
The other problem is that I use other methods, such as combat, removeBatcheffect, etc., to produce negative values, how to deal with these negative values? Will PLSDAbatch have the same problem?

**Comment by EvaYiwenWang** [comment]
May I know what kind of biological effect you would like to study after batch removal?  You cannot get rid of negative values because of the normalisation, like CLR. However, it doesn't matter for downstream analysis like differential expression analysis. 

Best,
Eva

**Comment by zhukun3** [comment]
> May I know what kind of biological effect you would like to study after batch removal? You cannot get rid of negative values because of the normalisation, like CLR. However, it doesn't matter for downstream analysis like differential expression analysis.
> 
> Best, Eva

I will do the differential expression analysis later, how to do it specifically?

**Comment by EvaYiwenWang** [comment]
There are two options: 1) Univariate methods: You can use simple linear regression (`lm()`) or a linear mixed model (`lmer()`, available in the lmerTest R package) if you need to handle random covariates. 2) Multivariate methods: PLSDA (available in the mixOmics R package). You can refer to the following case study: http://mixomics.org/case-studies/splsda-srbct-case-study/. I hope these help.

Best regards,
Eva

**Comment by zhukun3** [comment]
> May I know what kind of biological effect you would like to study after batch removal? You cannot get rid of negative values because of the normalisation, like CLR. However, it doesn't matter for downstream analysis like differential expression analysis.
> 
> Best, Eva

But how can I use negative values for differential expression analysis? Don't negative values represent expression levels of 0?

**Comment by EvaYiwenWang** [comment]
> > May I know what kind of biological effect you would like to study after batch removal? You cannot get rid of negative values because of the normalisation, like CLR. However, it doesn't matter for downstream analysis like differential expression analysis.
> > Best, Eva
> 
> But how can I use negative values for differential expression analysis? Don't negative values represent expression levels of 0?

No, because the values are centered and scaled, meaning the means are all adjusted to zero. Therefore, the negative values simply indicate how much lower they are compared to the mean.

**Comment by EvaYiwenWang** [comment]
In addition, the differential expression analysis only measures whether the differences between conditions (or groups of interest) are statistically significant. The focus is not on the value of an individual sample, but rather on the differences between the values of different groups of samples.

Best regards,
Eva
